tests/Functional/Driver/Mysqli/ResultTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\Mysqli;
+
+use Doctrine\DBAL\Driver\Mysqli\Result;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use mysqli;
+use mysqli_driver;
+use mysqli_sql_exception;
+
+use function assert;
+use function sprintf;
+
+use const MYSQLI_REPORT_ERROR;
+use const MYSQLI_REPORT_OFF;
+use const MYSQLI_REPORT_STRICT;
+
+/** @requires extension mysqli */
+final class ResultTest extends FunctionalTestCase
+{
+    private const TABLE_NAME = 'result_test_table';
+
+    private mysqli $nativeConnection;
+
+    protected function setUp(): void
+    {
+        if (! TestUtil::isDriverOneOf('mysqli')) {
+            self::markTestSkipped('This test requires the mysqli driver.');
+
+            return;
+        }
+
+        $nativeConnection = $this->connection->getNativeConnection();
+
+        assert($nativeConnection instanceof mysqli);
+
+        $this->nativeConnection = $nativeConnection;
+
+        $table = new Table(self::TABLE_NAME);
+        $table->addColumn('my_col_1', 'integer', ['notnull' => true]);
+
+        $this->dropAndCreateTable($table);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->dropTableIfExists(self::TABLE_NAME);
+    }
+
+    public function testSuccessfulRowCountFromAffectedRows(): void
+    {
+        $mysqliStmt = $this->nativeConnection
+            ->prepare(sprintf('INSERT INTO %s VALUES (7);', self::TABLE_NAME));
+        $mysqliStmt->execute();
+
+        self::assertSame(1, $mysqliStmt->affected_rows);
+        self::assertSame(1, (new Result($mysqliStmt))->rowCount());
+    }
+
+    public function testFailingRowCountFromAffectedRows(): void
+    {
+        $mysqliStmt = $this->nativeConnection
+            ->prepare(sprintf('INSERT INTO %s VALUES (NULL);', self::TABLE_NAME));
```

Comment:
I don't think we need to use the native APIs here. The very same should happen if we use the DBAL API for preparing/executing the statement, so let's do that instead.

Reply:
I'm using the native API to build the `Result` object in an explicit and isolated way, in order to keep this test narrower as possible regarding the tested components.